services/mailer/mail_workflow_run.go

```diff
@@ -0,0 +1,166 @@
+	metadataHeaders := generateMetadataHeaders(repo)
+
+	jobs, err := actions_model.GetRunJobsByRunID(ctx, run.ID)
+	if err != nil {
+		log.Error("GetRunJobsByRunID: %v", err)
+		return
+	}
+	sort.SliceStable(jobs, func(i, j int) bool {
+		si, sj := jobs[i].Status, jobs[j].Status
+		/*
+			If both i and j are/are not success, leave it to si < sj.
+			If i is success and j is not, since the desired is j goes "smaller" and i goes "bigger", this func should return false.
+			If j is success and i is not, since the desired is i goes "smaller" and j goes "bigger", this func should return true.
+		*/
+		if si.IsSuccess() != sj.IsSuccess() {
+			return !si.IsSuccess()
+		}
+		return si < sj
+	})
+
+	convertedJobs := make([]convertedWorkflowJob, 0, len(jobs))
+	for _, job := range jobs {
+		converted0, err := convert.ToActionWorkflowJob(ctx, repo, nil, job)
+		if err != nil {
+			log.Error("convert.ToActionWorkflowJob: %v", err)
+			continue
+		}
+		convertedJobs = append(convertedJobs, convertedWorkflowJob{
+			HTMLURL: converted0.HTMLURL,
+			Name:    converted0.Name,
+			Status:  job.Status,
+			Attempt: converted0.RunAttempt,
+		})
+	}
+
+	langMap := make(map[string][]*user_model.User)
+	for _, user := range recipients {
+		langMap[user.Language] = append(langMap[user.Language], user)
+	}
+	for lang, tos := range langMap {
+		locale := translation.NewLocale(lang)
+		var runStatusText string
+		switch run.Status {
+		case actions_model.StatusSuccess:
+			runStatusText = "All jobs have succeeded"
+		case actions_model.StatusFailure:
+			runStatusText = "All jobs have failed"
+			for _, job := range jobs {
+				if !job.Status.IsFailure() {
+					runStatusText = "Some jobs were not successful"
+					break
+				}
+			}
+		case actions_model.StatusCancelled:
+			runStatusText = "All jobs have been cancelled"
+		}
+		var mailBody bytes.Buffer
+		if err := LoadedTemplates().BodyTemplates.ExecuteTemplate(&mailBody, tplWorkflowRun, map[string]any{
+			"Subject":       subject,
+			"Repo":          repo,
+			"Run":           run,
+			"RunStatusText": runStatusText,
+			"Jobs":          convertedJobs,
+			"locale":        locale,
+			"Language":      locale.Language(),
+		}); err != nil {
+			log.Error("ExecuteTemplate [%s]: %v", tplWorkflowRun, err)
+			return
+		}
+		msgs := make([]*sender_service.Message, 0, len(tos))
+		for _, rec := range tos {
+			msg := sender_service.NewMessageFrom(
+				rec.Email,
+				displayName,
+				setting.MailService.FromEmail,
+				subject,
+				mailBody.String(),
+			)
+			msg.Info = subject
+			for k, v := range generateSenderRecipientHeaders(sender, rec) {
+				msg.SetHeader(k, v)
+			}
+			for k, v := range metadataHeaders {
+				msg.SetHeader(k, v)
+			}
+			msg.SetHeader("Message-ID", messageID)
+			msgs = append(msgs, msg)
+		}
+		SendAsync(msgs...)
+	}
+}
+
+func SendActionsWorkflowRunStatusEmail(ctx context.Context, sender *user_model.User, repo *repo_model.Repository, run *actions_model.ActionRun) {
```

Comment:
Why there are two `SendActionsWorkflowRunStatusEmail` `sendActionsWorkflowRunStatusEmail` for different purposes ...

Is it possible to make them have different names?